spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/store/ConfigurableMongoDbMessageGroupStoreTests.java

```diff
@@ -134,6 +138,11 @@ void testWithCustomConverter() {
 						.getClass());
 		context.refresh();
 
+		MongoTemplate mongoTemplate = new MongoTemplate(MONGO_DATABASE_FACTORY);
+		List<IndexInfo> indexesInfo = mongoTemplate.indexOps(MongoDbChannelMessageStore.DEFAULT_COLLECTION_NAME).getIndexInfo();
+
+		assertThat(indexesInfo.size()).isEqualTo(4);
```

Comment:
See my comment in the issue about this problem: https://github.com/spring-projects/spring-integration/issues/8692#issuecomment-1696156878.
So, apparently we need to remove indexes on that collection before creating an application context in this test.
Or we can just remove collection altogether.

There is also a nice fluent API in AssertJ for collections:
```
assertThat(indexesInfo).hasSize(0);
```